amethyst_animation/src/bundle.rs

```diff
@@ -1,53 +1,34 @@
 use std::{hash::Hash, marker};
 
-use amethyst_core::{
-    ecs::prelude::{Component, DispatcherBuilder, World},
-    SystemBundle, SystemDesc,
-};
+use amethyst_core::ecs::*;
 use amethyst_error::Error;
+use derivative::Derivative;
+use marker::PhantomData;
 
-use crate::{
-    resources::AnimationSampling,
-    skinning::VertexSkinningSystemDesc,
-    systems::{
-        AnimationControlSystemDesc, AnimationProcessor, SamplerInterpolationSystem,
-        SamplerProcessor,
-    },
-};
+use crate::resources::AnimationSampling;
 
 /// Bundle for vertex skinning
 ///
 /// This registers `VertexSkinningSystem`.
 /// Note that the user must make sure this system runs after `TransformSystem`
 #[derive(Default, Debug)]
-pub struct VertexSkinningBundle<'a> {
-    dep: &'a [&'a str],
-}
+pub struct VertexSkinningBundle;
 
-impl<'a> VertexSkinningBundle<'a> {
+impl VertexSkinningBundle {
     /// Create a new sampling bundle
     pub fn new() -> Self {
         Default::default()
     }
-
-    /// Set dependencies for the `VertexSkinningSystem`
-    pub fn with_dep(mut self, dep: &'a [&'a str]) -> Self {
-        self.dep = dep;
-        self
-    }
 }
 
-impl<'a, 'b, 'c> SystemBundle<'a, 'b> for VertexSkinningBundle<'c> {
-    fn build(
-        self,
-        world: &mut World,
-        builder: &mut DispatcherBuilder<'a, 'b>,
+impl SystemBundle for VertexSkinningBundle {
+    fn load(
+        &mut self,
+        _world: &mut World,
+        _resources: &mut Resources,
+        _builder: &mut DispatcherBuilder,
     ) -> Result<(), Error> {
-        builder.add(
-            VertexSkinningSystemDesc::default().build(world),
-            "vertex_skinning_system",
-            self.dep,
-        );
+        // FIXME: builder.add_system(VertexSkinningSystem);
```

Comment:
What is missing for this?

Reply:
I think i've converted it now, but we won't be able to test until the gltf example is working.